File: server/apps/assignments/application/dto/assignment_list_dto.py
from dataclasses import dataclass
from decimal import Decimal
from typing import Optional
from datetime import datetime

@dataclass
class AssignmentListDTO:
    assignment_id: str
    subject_name: str
    title: str
    description: Optional[str]
    assignment_type: str
    total_score: Decimal
    status: str
    is_active: bool
    created_at: datetime

File: client/src/application/assignments/dto/assignment_list.dto.js
class AssignmentListDTO {
  constructor({
    assignment_id,
    subject_name,
    title,
    description = null,
    assignment_type,
    total_score,
    status,
    is_active,
    created_at,
  }) {
    this.assignmentId = String(assignment_id || "");
    this.subjectName = String(subject_name || "");

    this.title = String(title || "");

    this.description = typeof description === "string" ? description : null;

    this.assignmentType = String(assignment_type || "");

    this.totalScore = total_score;

    this.status = String(status || "");

    this.isActive = Boolean(is_active);

    this.createdAt = created_at;
  }

  static fromResponse(data = {}) {
    return new AssignmentListDTO(data);
  }
}

module.exports = AssignmentListDTO;
